cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -201,6 +201,108 @@ func (c *Commander) watch() {
 	c.exitCh <- struct{}{}
 }
 
+// StartOneShot starts the Collector with the expectation that it will immediately
+// exit after it finishes a quick operation. This is useful for situations like reading stdout/sterr
+// to e.g. check the feature gate the Collector supports.
+func (c *Commander) StartOneShot() ([]byte, []byte, error) {
```

Comment:
This function shares a lot of similar code to some of the other Commander methods, but I couldn't find a good way to share the code between them all in a way that made sense for only two instances of the functionality. I think we can revisit trying to share some of this logic in helper methods if we find a need for a third way to execute the Collector in the future.